libs/ai-endpoints/langchain_nvidia_ai_endpoints/embeddings.py

```diff
@@ -35,48 +41,67 @@ class NVIDIAEmbeddings(_NVIDIAClient, Embeddings):
             "Default is 'NONE', which raises an error if an input is too long."
         ),
     )
-    max_length: int = Field(2048, ge=1, le=2048)
     max_batch_size: int = Field(default=_default_max_batch_size)
     model_type: Optional[Literal["passage", "query"]] = Field(
-        None, description="The type of text to be embedded."
+        None, description="(DEPRECATED) The type of text to be embedded."
     )
 
-    # indicate to user that max_length is deprecated when passed as an argument to
-    # NVIDIAEmbeddings' constructor, e.g. NVIDIAEmbeddings(max_length=...). this
-    # does not warning on assignment, e.g. embedder.max_length = ...
-    # todo: fix _NVIDIAClient.validate_client and enable Config.validate_assignment
-    @validator("max_length")
-    def deprecated_max_length(cls, value: int) -> int:
-        """Deprecate the max_length field."""
-        warnings.warn(
-            "The max_length field is deprecated. Use the 'truncate' instead.",
-            DeprecationWarning,
+    def __init__(self, **kwargs: Any):
+        super().__init__(**kwargs)
+        infer_path = "{base_url}/embeddings"
+        # not all embedding models are on https://integrate.api.nvidia.com/v1,
+        # those that are not are served from their own endpoints
+        if model := determine_model(self.model):
+            if model.endpoint:  # some models have custom endpoints
+                infer_path = model.endpoint
+        self._client = _NVIDIAClient(
+            base_url=self.base_url,
+            model=self.model,
+            api_key=kwargs.get("nvidia_api_key", kwargs.get("api_key", None)),
+            infer_path=infer_path,
         )
-        return value
-
-    # todo: fix _NVIDIAClient.validate_client and enable Config.validate_assignment
-    @validator("model")
-    def aifm_deprecated(cls, value: str) -> str:
-        """All AI Foundataion Models are deprecate, use API Catalog models instead."""
-        for model in [value, f"playground_{value}"]:
-            if model in MODEL_SPECS and MODEL_SPECS[model].get("api_type") == "aifm":
-                alternative = MODEL_SPECS[model].get(
-                    "alternative", NVIDIAEmbeddings._default_model
-                )
-                warnings.warn(
-                    f"{value} is deprecated. Try {alternative} instead.",
-                    DeprecationWarning,
-                )
-        return value
+        # todo: only store the model in one place
+        # the model may be updated to a newer name during initialization
+        self.model = self._client.model
+
+        # todo: remove when nvolveqa_40k is removed from MODEL_TABLE
+        if "model" in kwargs and kwargs["model"] in [
```

Comment:
Is this to validate the original model user input? why checking from kwargs?

Reply:
yes, checking the original model input by the user. at this point in the code self.model will be a transformed version of the original input, e.g. nvolveqa_40k -> NV-Embed-QA.